compiler/passes/normalize.cpp

```diff
@@ -1751,14 +1751,15 @@ static void change_method_into_constructor(FnSymbol* fn) {
 
   // Now check that the function name matches the name of the type
   // attached to 'this'.
```

Comment:
Yes, I think that if we declared our own string type (that amounts to const char* for now, but ensures that the value has been unique'd with astr) we could easily support == and even support it in the future if we changed our mind about the giant string hashtable uniquing all strings.